lib/rules/jsx-curly-brace-presence.js

```diff
@@ -205,17 +205,44 @@ module.exports = {
       );
     }
 
-    function shouldCheckForUnnecessaryCurly(parent, config) {
-      // If there are more than one JSX child, there is no need to check for
-      // unnecessary curly braces.
-      if (jsxUtil.isJSX(parent) && parent.children.length !== 1) {
+    function isWhiteSpaceLiteral(node) {
+      return node.type && node.type === 'Literal' && node.value && jsxUtil.isWhiteSpaces(node.value);
+    }
+
+    function getAdjacentSiblings(node, children) {
+      for (let i = 1; i < children.length - 1; i++) {
+        const child = children[i];
+        if (node === child) {
+          return [children[i - 1], children[i + 1]];
+        }
+      }
+      if (node === children[0] && children[1]) {
+        return [children[1]];
+      }
+      if (node === children[children.length - 1] && children[children.length - 2]) {
+        return [children[children.length - 2]];
+      }
+      return [];
+    }
+
+    function hasAdjacentJsxExpressionContainers(node, children) {
+      const childrenExcludingWhitespaceLiteral = children.filter(child => !isWhiteSpaceLiteral(child));
+      const adjSiblings = getAdjacentSiblings(node, childrenExcludingWhitespaceLiteral);
```

Comment:
I don't think this check is quite right. We now get stuff like:

```diff
           <div>
-            Confirm{' '}
+            Confirm
             <button type="button" onClick={this.onClickYes}>
```

But this is not the same markup and it's not desirable here.